storage/src/main/java/tech/pegasys/artemis/storage/CombinedChainDataClient.java

```diff
@@ -146,6 +146,11 @@ public CombinedChainDataClient(
     return getBlockByBlockRoot(get_block_root_at_slot(state, slot));
   }
 
+  private boolean isHistoricalData(final UnsignedLong slot) {
+    final boolean finalizedPastFirstEpoch = !recentChainData.getFinalizedEpoch().equals(ZERO);
```

Comment:
We always hold the latest finalized state in the in-memory store.  What if we just check the `chainStorageClient` first before pulling historical data?

Reply:
To find state we wind up having to walk backwards through the historical roots so it's a significant advantage to know that something is finalised and go straight to the historical store where we can just lookup by slot.